src/pages/patch/ActionButtons.tsx

```diff
@@ -1,70 +1,129 @@
-import { useOnClickOutside } from "hooks";
+import { useParams } from "react-router-dom";
 import { InputNumber, Popconfirm } from "antd";
-import get from "lodash/get";
+import { useMutation } from "@apollo/react-hooks";
 import { Body } from "@leafygreen-ui/typography";
 import { PageButtonRow } from "components/styles";
 import { DropdownItem, ButtonDropdown } from "components/ButtonDropdown";
 import { PatchRestartModal } from "pages/patch/index";
+import { useBannerDispatchContext } from "context/banners";
+import {
+  SchedulePatchTasksMutation,
+  SchedulePatchTasksMutationVariables,
+  UnschedulePatchTasksMutation,
+  UnschedulePatchTasksMutationVariables,
+  SetPatchPriorityMutation,
+  SetPatchPriorityMutationVariables,
+} from "gql/generated/types";
+import {
+  SCHEDULE_PATCH_TASKS,
+  UNSCHEDULE_PATCH_TASKS,
+  SET_PATCH_PRIORITY,
+} from "gql/mutations";
 
-interface Props {
-  initialPriority?: number;
-  canAbort: boolean;
-  canRestart: boolean;
-  canSchedule: boolean;
-  canUnschedule: boolean;
-  canSetPriority: boolean;
-}
-
-export const ActionButtons = ({
-  canAbort,
-  canRestart,
-  canSchedule,
-  canSetPriority,
-  canUnschedule,
-  initialPriority,
-}: Props) => {
+export const ActionButtons = () => {
+  const { successBanner, errorBanner } = useBannerDispatchContext();
   const wrapperRef = useRef(null);
   const priorityRef = useRef(null);
   const [isVisible, setIsVisible] = useState(false);
   const [openModal, setOpenModal] = useState(false);
+  const [abort, setAbort] = useState(false);
+  const initialPriority = 1;
   const [priority, setPriority] = useState<number>(initialPriority);
+  const { id: patchId } = useParams<{ id: string }>();
+
+  const [
+    schedulePatchTasks,
+    { loading: loadingSchedulePatchTasks },
+  ] = useMutation<
+    SchedulePatchTasksMutation,
+    SchedulePatchTasksMutationVariables
+  >(SCHEDULE_PATCH_TASKS, {
+    variables: { patchId },
+    onCompleted: () => {
+      successBanner("All tasks were scheduled");
+    },
+    onError: (err) => {
+      errorBanner(`Error scheduling tasks: ${err.message}`);
+    },
+    refetchQueries,
+  });
+
+  const [
+    unschedulePatchTasks,
+    { loading: loadingUnschedulePatchTasks },
+  ] = useMutation<
+    UnschedulePatchTasksMutation,
+    UnschedulePatchTasksMutationVariables
+  >(UNSCHEDULE_PATCH_TASKS, {
+    onCompleted: () => {
+      successBanner(
+        `All tasks were unscheduled ${
+          abort ? "and tasks that already started were aborted" : ""
+        }`
+      );
+      setAbort(false);
+    },
+    onError: (err) => {
+      errorBanner(`Error unscheduling tasks: ${err.message}`);
+    },
+    refetchQueries,
+  });
+
+  const [setPatchPriority, { loading: loadingSetPatchPriority }] = useMutation<
+    SetPatchPriorityMutation,
+    SetPatchPriorityMutationVariables
+  >(SET_PATCH_PRIORITY, {
+    onCompleted: () => {
+      successBanner(`Priority was set to ${priority}`);
+      setPriority(initialPriority);
```

Comment:
I think we should keep the priority set to the new value. This way the user knows the current value is before changing and this is how the modal currently works in the EVG ui.

Reply:
totes agree. done.